app/views/posts/index.html.erb

```diff
@@ -2,11 +2,12 @@
 <%= render partial:'shared/user_header', locals:{user: User.find(params[:user_id])} %>
 <% @posts.each do |post| %>
    <%= render partial:'shared/post', locals:{post: post} %> 
-   <% post.comments.each do |comment|%>
```

Comment:
- [ ] Kindly solve the` N+1` problem when fetching all posts and their comments for a user.
Example in the `posts#index` actions where you're fetching all posts for a user, please use `@posts = Post.where(author_id: params[:user_id]).includes(:comments)` instead of `    @posts = Post.where(author_id: params[:user_id])` This will solve `N + 1` query problem because you will be fetching the comments while fetching the posts for a user and the number of query reduces.

Reply:
Thank you @julie-ify, I tried doing this but I got an error from bullet.log.
`2022-07-27 14:59:36[WARN] user: USER
GET /users/1/posts
AVOID eager loading detected
  Post => [:comments]
  Remove from your query: .includes([:comments])
Call stack`

Reply:
@tylher, If that's the case, leave it the way it was and submit it again 😃 since there is a call stack.

Reply:
Okay thank you very much